src/edu/csus/ecs/pc2/ui/MessageManager.java

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 1989-2023 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.ui;
+
+import java.util.Vector;
+
+/**
+ * Manage a set of message listeners.
+ * 
+ * @author Douglas A. Lane <pc2@ecs.csus.edu>
+ *
+ */
+public class MessageManager {
```

Comment:
Frankly, I don't think this class, or the related classes such as `IMessageRecordListener`, belong in this PR; they have little or nothing to do with the stated purpose of the PR ("fix load samples find samples in executable").  These classes should have been submitted as a separate Issue/PR.

However, what's done is done and I certainly don't, at this point, want to suggest we go back and redo everything.  I just want to point out that I don't think it was properly done to begin with; it's very hard to justify something so big (multiple classes; around 200 lines of code) as being just "continuous improvement" or that it rightly belongs in this PR...